enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * PowerAuth Enrollment Server
+ * Copyright (C) 2022 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.enrollmentserver.impl.service.converter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wultra.app.enrollmentserver.database.entity.OperationTemplateEntity;
+import com.wultra.security.powerauth.client.model.enumeration.OperationStatus;
+import com.wultra.security.powerauth.client.model.enumeration.SignatureType;
+import com.wultra.security.powerauth.client.model.response.OperationDetailResponse;
+import com.wultra.security.powerauth.lib.mtoken.model.entity.Operation;
+import com.wultra.security.powerauth.lib.mtoken.model.entity.PreApprovalScreen;
+import com.wultra.security.powerauth.lib.mtoken.model.entity.UiExtensions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test for {@link MobileTokenConverter}.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+class MobileTokenConverterTest {
+
+    private MobileTokenConverter tested = new MobileTokenConverter(new ObjectMapper());
+
+    @Test
+    void testConvertUi_null() throws Exception {
+        final OperationDetailResponse operationDetail = new OperationDetailResponse();
+        operationDetail.setSignatureType(List.of(SignatureType.KNOWLEDGE));
+        operationDetail.setStatus(OperationStatus.APPROVED);
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+
+        final Operation result = tested.convert(operationDetail, operationTemplate);
+
+        assertNull(result.getUi());
+    }
+
+    @Test
+    void testConvertUi_overriddenByEnrollment() throws Exception {
+        final OperationDetailResponse operationDetail = new OperationDetailResponse();
+        operationDetail.setSignatureType(List.of(SignatureType.KNOWLEDGE));
+        operationDetail.setStatus(OperationStatus.APPROVED);
+        operationDetail.setRiskFlags("C");
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+        operationTemplate.setUi("{\n" +
```

Comment:
We do not want to use Czech language in upstream. The String should be moved into a constant.

Reply:
Understood, I will change it. I took the template from the issue, where it is in Czech.